src/Classes/Particle/Attracter.ts

```diff
@@ -2,14 +2,16 @@ import type { IParticle } from "../../Interfaces/IParticle";
 import type { Container } from "../Container";
 
 export class Attracter {
-	public static attract(p1: IParticle, p2: IParticle, container: Container): void {
+	public static attract(p1: IParticle, container: Container): void {
 		const options = container.options;
-		/* condensed particles */
-		const dx = p1.position.x - p2.position.x;
-		const dy = p1.position.y - p2.position.y;
-		const dist = Math.sqrt(dx * dx + dy * dy);
 
-		if (dist <= (p1.lineLinkedDistance ?? container.retina.lineLinkedDistance)) {
+		for (const p2 of container.particles.spatialGrid.queryRadius(p1.position, container.retina.lineLinkedDistance)) {
```

Comment:
container.retina.lineLinkedDistance should be (p1.lineLinkedDistance ?? container.retina.lineLinkedDistance) or something like that. p1 and p2 could have a different distance 🤔